spec/integration/provider/mount_spec.rb

```diff
@@ -123,27 +106,31 @@ def run_in_catalog(settings)
             end
             expected_fstab_data = (ensure_setting != :absent)
             describe "When setting ensure => #{ensure_setting}" do
-              ["local", "journaled"].each do |options_setting|
-                describe "When setting options => #{options_setting}" do
+              ["local", "journaled", "", nil].each do |options_setting|
+                describe "When setting options => '#{options_setting}'" do
                   it "should leave the system in the #{expected_final_state ? 'mounted' : 'unmounted'} state, #{expected_fstab_data ? 'with' : 'without'} data in /etc/fstab" do
                     if family == "Solaris"
                       skip("Solaris: The mock :operatingsystem value does not get changed in lib/puppet/provider/mount/parsed.rb")
                     else
-                      @desired_options = options_setting
-                      run_in_catalog(:ensure=>ensure_setting, :options => options_setting)
-                      expect(@mounted).to eq(expected_final_state)
-                      if expected_fstab_data
-                        expect(check_fstab(expected_fstab_data)).to eq("/dev/disk1s1")
+                      if options_setting && options_setting.empty?
+                        expect { run_in_catalog(:ensure=>ensure_setting, :options => options_setting) }.to raise_error Puppet::ResourceError
```

Comment:
@haus @Sharpie and I decided it wasn't worth the additional nesting and magic to add that test case. Before these changes, this test really wasn't doing much at all that was helpful, since it was only testing darwin. The resource error here does catch the error raised on the line you commented above.